conans/client/api/conan_api.py

```diff
@@ -239,39 +241,120 @@ def search_remote_recipes(self, query, remote):
+    def get_package_revisions(self, reference, remote=None):
+        """
+        Get all the recipe revisions given a reference from cache or remote.
+
+        :param reference: `PackageReference` without the revision
+        :param remote: `Remote` object
+        :return: `dict` with all the results, e.g.,
+                 {"remote": "my_remote_name",  # or None
+                  "reference": "libyaml/0.2.5#80b7cbe095ac7f38844b6511e69e453a:ef93ea55bee154729e264db35ca6a16ecab77eb7",
+                  "results": [{"revision": "80b7cbe095ac7f38844b6511e69e453a",
+                               "time": "2021-07-20 00:56:25 UTC"}, ...]}
+        """
+        return self._get_revisions(reference, 'package', remote=remote)
 
-        recipe_revisions = self.app.cache.get_recipe_revisions(ref)
-        results = []
-        for revision in recipe_revisions:
-            timestamp = self.app.cache.get_timestamp(revision)
-            result = {
-                "revision": revision.revision,
-                "time": from_timestamp_to_iso8601(timestamp)
-            }
-            results.append(result)
-        return results
+    @api_method
+    def get_recipe_revisions(self, reference, remote=None):
+        """
+        Get all the recipe revisions given a reference from cache or remote.
+
+        :param reference: `ConanFileReference` without the revision
+        :param remote: `Remote` object
+        :return: `dict` with all the results, e.g.,
+                 {"remote": "my_remote_name",  # or None
+                  "reference": "libyaml/0.2.5",
+                  "results": [{"revision": "80b7cbe095ac7f38844b6511e69e453a",
+                               "time": "2021-07-20 00:56:25 UTC"}, ...]}
+        """
+        return self._get_revisions(reference, 'recipe', remote=remote)
+
+    @api_method
+    def get_package_ids(self, reference, remote=None):
+        """
+        Get all the Package IDs given a recipe revision from cache or remote.
+
+        Note: if reference does not have the revision, we'll return the Package IDs for
+        the latest recipe revision by default
+
+        :param reference: `ConanFileReference` with/without revision
+        :param remote: `Remote` object
+        :return: `list` of `dict` with the `package-id` as keys, e.g.,
+                 {"remote": "my_remote_name",  # or None
+                  "reference": "libcurl/7.77.0#2a9c4fcc8d76d891e4db529efbe24242",
+                  "results": {"d5f16437dd4989cc688211b95c24525589acaafd": {
+                                    "settings": {"compiler": "apple-clang",...},
+                                    "options": {'options': {'shared': 'False',...}},
+                                    "requires": [
+                                        'mylib/1.0.8:3df6ebb8a308d309e882b21988fd9ea103560e16',...]}
+                            }
+                 }
+        """
+        ref = ConanFileReference.loads(reference)
```

Comment:
When this gets too heavy on business logic, we should delegate the code to helper files.
No need to change this here, we will do it when we check the whole api.